GOVERNANCE.md

```diff
@@ -0,0 +1,118 @@
+# Governance thin-edge.io
+
+This documents servers as a guiding document to ease technical decision making in the project as a whole.
+
+Some of its goals are:
+
+- Making sure that the different stakeholders are represented with their requirements
+- Making sure that contributors know how decisions are made and who they can address concerns/questions to
+- Making sure that team members know what is required of them for the project
+
+
+## The project vision
+
+In the [vision.md](./vision.md) the goal of the project is encoded.
+It specifies the different functional requirements and non-functional requirements of the project as a whole.
+All the stakeholders must agree on the vision upon joining.
+
+- The current stakeholders are: Software AG, and IFM
+
+Changes to the vision need to be done in a plenum of stakeholders, this makes sure that everyone is aware and agrees to the evolution of the project.
+
+
+
+## Team Structure
+
+* Teams are either single contributors or a group of them
+* They are responsible for a specific part of project, and as such are the maintainers of those parts
+    * This includes: Reviewing pull requests, triaging issuess assigned to them, and general maintenance
+* Each team is assigned a single or multiple subtrees of the project, as defined by the project structure
```

Comment:
Question: I like to believe that assigned subtrees might overlap. Should we explicitely state here that assigned subtrees of teams might overlap?

Reply:
By "assigned subtrees of teams might overlap", do you mean that a subtree might be managed by more than one team? If so this is what I expect too, since some component/plugin/subtree might be used by many and therefore need specific coordination.

We have to improve the wording here.

Reply:
> By "assigned subtrees of teams might overlap", do you mean that a subtree might be managed by more than one team?

Yes, at least partially. An (maybe far-fetched, maybe not...) example would be a team that manages `plugins/plugin_a/*` and a team that takes care of dependencies and manages all `Cargo.toml` files, so also `plugins/plugin_a/Cargo.toml` - this would be an overlap. The team for the plugin is of course able to change the `Cargo.toml` of the plugin, but the "dependency team" might change it for some reason.

A simple addition would suffice here, I guess.
Maybe something like "subtrees might be (partially or fully) managed by one than more team" - with emphasis on "might".